### PR TITLE
Fix some issues found by static analysis

### PR DESCRIPTION
**Pull request by vlstill**

### CODE REVIEW

**Review by fruffy — DISMISSED** [review]
Which linter is this? Publicly available?

**Comment by vlstill** [comment]
> Which linter is this? Publicly available?

Coverity, so no. My feeling is a lot of this could actually be found by clang-tidy too though, but I had not checked that yet.

File: backends/p4tools/modules/testgen/targets/ebpf/test_backend.cpp [hunk 1]
```diff
@@ -66,12 +66,14 @@ TestBackEnd::TestInfo EBPFTestBackend::produceTestInfo(
             IR::getConstant(IR::getBitType(outPktSize), EBPFTestBackend::ZERO_PKT_VAL);
         testInfo.packetTaintMask =
             IR::getConstant(IR::getBitType(outPktSize), EBPFTestBackend::ZERO_PKT_MAX);
+    } else {
```
[the remark this answers is not in the document]
Comment (fruffy) [reply backends/p4tools/modules/testgen/targets/ebpf/test_backend.cpp:69]: This seems good. I am not even sure whether the condition can ever be true. The eBPF back end can not reduce packet size.

File: ir/v1.def [hunk 2]
```diff
@@ -230,7 +230,7 @@ class CalculatedField : IAnnotated {
         bool                    update;
         ID                      name;
         Expression              cond;
-        update_or_verify() {}  // FIXME -- needed by umpack_json(safe_vector) -- should not be
+        update_or_verify() { update = false; }  // FIXME -- needed by umpack_json(safe_vector) -- should not be
```
Comment (fruffy) [inline ir/v1.def:233]: Should this be in the function call or in the initializer of `update`?

Comment (vlstill) [reply ir/v1.def:233]: I would like to put it into the initializer, but for some reason that breaks the generated code -- the parametric constructor has just 3 arguments. I don't know why that is, it seems quite weird to me. Therefore I've added it to the constructor. I'm quite surprised nested classes can be generated by IRGen.

Comment (fruffy) [reply ir/v1.def:233]: `optional bool           update = false;`
maybe?

**Comment by vlstill** [comment]
@fruffy is this test working normally?
image
I've seen in failed also in

**Comment by fruffy** [comment]
> @fruffy is this test working normally? image I've seen in failed also in

It is failing about 30% of the time. We are unsure what causes this failure but are trying to fix it. It might be caused by a particular Github runner being chosen.

File: ir/v1.def (hunk 2 above)
Comment (vlstill) [reply ir/v1.def]: That worked, thanks.

**Pull request merged**